newIDE/app/src/EventsSheet/InstructionEditor/EventTextDialog.js

```diff
@@ -0,0 +1,240 @@
+// @flow
+import { Trans } from '@lingui/macro';
+import { t } from '@lingui/macro';
+
+import * as React from 'react';
+import Dialog from '../../UI/Dialog';
+import FlatButton from '../../UI/FlatButton';
+import HelpButton from '../../UI/HelpButton';
+import { Line, Column } from '../../UI/Grid';
+import ColorPicker from '../../UI/ColorField/ColorPicker';
+import MiniToolbar, { MiniToolbarText } from '../../UI/MiniToolbar';
+import SemiControlledTextField from '../../UI/SemiControlledTextField';
+
+const gd = global.gd;
+
+const toolbarItemStyle = {
+  marginRight: 10,
+};
+
+const styles = {
+  sizeTextField: {
+    width: 90,
+    ...toolbarItemStyle,
+  },
+  toolbarItem: toolbarItemStyle,
+  checkbox: toolbarItemStyle,
+};
+type RGBColor = {|
+  r?: number,
+  g?: number,
+  b?: number,
+  a?: number,
+|};
+
+type Props = {|
+  event: gdBaseEvent,
+  onClose: () => void,
+  onApply: () => void,
+|};
+
+type State = {|
+  textValue: string,
+  textColor: RGBColor,
+  backgroundColor: RGBColor,
+|};
+
+export default class EventTextDialog extends React.Component<Props, State> {
+  _applyChangesOnEvent = () => {
+    const { event } = this.props;
+    const { textValue, textColor, backgroundColor } = this.state;
+    const eventType = event.getType();
+
+    if (eventType === 'BuiltinCommonInstructions::Comment') {
+      //Text value
+      gd.asCommentEvent(event).setComment(textValue);
+
+      //Text color
+      gd.asCommentEvent(event).setTextColor(
+        parseInt(textColor.r),
+        parseInt(textColor.g),
+        parseInt(textColor.b)
+      );
+      //Background color
+      gd.asCommentEvent(event).setBackgroundColor(
+        parseInt(backgroundColor.r),
+        parseInt(backgroundColor.g),
+        parseInt(backgroundColor.b)
+      );
+    } else if (eventType === 'BuiltinCommonInstructions::Group') {
+      //Text value
+      gd.asGroupEvent(event).setName(textValue);
+
+      //Text color for group not supported in Core, instead GroupEvent.js handle this
+      //Background color
+      gd.asGroupEvent(event).setBackgroundColor(
+        parseInt(backgroundColor.r),
+        parseInt(backgroundColor.g),
+        parseInt(backgroundColor.b)
+      );
+      return;
+    }
+  };
+
+  _getInitialStateFromEvent = () => {
+    const { event } = this.props;
+    const eventType = event.getType();
+    let text: string, textColors: RGBColor, backgroundColors: RGBColor;
```

Comment:
This is kind of "false" as you're declaring variables to be of a certain type, but they are actually undefined :) 
So either you mark them as optional (but don't do that) or (better) you initialize them to default values:

```
let text: string = "";
let textColors: RGBColor = {r:0, g:0,b:0, a:0};
let backgroundColors: RGBColor = {r:0, g:0,b:0, a:0};
```

Otherwise, Flow will warn you: you might be using a variable that is uninitialized! (i.e.: that is undefined).
![image](https://user-images.githubusercontent.com/1280130/70317166-23124080-1815-11ea-961f-adccbb741d48.png)

Reply:
See my next comment about rename variables too to give them the same name as in the State.